pkg/k8scontext/k8scontext_test.go

```diff
@@ -437,8 +439,188 @@ var _ = ginkgo.Describe("K8scontext", func() {
+			finalList := ctxt.filterAndSort(ingrList)
 			Expect(finalList).To(ContainElement(ingr))
 		})
 	})
+
+	ginkgo.Context("Check IsIstioGatewayIngress", func() {
+		ginkgo.BeforeEach(func() {
+			ctxt.ingressClassControllerName = environment.DefaultIngressClassController
+		})
+
+		ginkgo.It("returns error when gateway has no annotations", func() {
+			gateway := &v1alpha3.Gateway{}
+			actual := ctxt.IsIstioGatewayIngress(gateway)
+			Expect(actual).To(Equal(false))
+		})
+
+		ginkgo.It("returns true with correct annotation", func() {
+			gateway := &v1alpha3.Gateway{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						annotations.IstioGatewayKey: environment.DefaultIngressClassController,
+					},
+				},
+			}
+			actual := ctxt.IsIstioGatewayIngress(gateway)
+			Expect(actual).To(Equal(true))
+		})
+	})
+
+	ginkgo.Context("Check IsApplicationGatewayIngress", func() {
+		ginkgo.BeforeEach(func() {
+			ctxt.ingressClassControllerName = environment.DefaultIngressClassController
+		})
+
+		ginkgo.It("returns error when ingress has no annotations", func() {
+			ing := &networking.Ingress{}
+			actual := ctxt.IsIngressClass(ing)
+			Expect(actual).To(Equal(false))
+		})
+
+		ginkgo.It("returns true with correct annotation", func() {
+			actual := ctxt.IsIngressClass(ingress)
+			Expect(actual).To(Equal(true))
+		})
+
+		ginkgo.It("returns true with correct annotation", func() {
+			ingress.Annotations[annotations.IngressClassKey] = "custom-class"
+			ctxt.ingressClassControllerName = "custom-class"
+			actual := ctxt.IsIngressClass(ingress)
+			Expect(actual).To(Equal(true))
+		})
+
+		ginkgo.It("returns false with incorrect annotation", func() {
+			ingress.Annotations[annotations.IngressClassKey] = "custom-class"
+			actual := ctxt.IsIngressClass(ingress)
+			Expect(ctxt.ingressClassControllerName).To(Equal(environment.DefaultIngressClassController))
+			Expect(actual).To(Equal(false))
+		})
+	})
+
+	ginkgo.Context("Check Ingress Class Resource is used correctly for filtering ingress", func() {
+		ginkgo.BeforeEach(func() {
+			// create and ingress class in k8s
+			ingressClass := tests.GetIngressClass()
+			ctxt.kubeClient.NetworkingV1().IngressClasses().Create(context.TODO(), ingressClass, metav1.CreateOptions{})
+
+			// update ingress resource to use ingress class name
+			delete(ingress.Annotations, annotations.IngressClassKey)
+			ingress.Spec.IngressClassName = to.StringPtr(environment.DefaultIngressClassResourceName)
+			_, err := k8sClient.NetworkingV1().Ingresses(ingressNS).Update(context.TODO(), ingress, metav1.UpdateOptions{})
+			Expect(err).ToNot(HaveOccurred(), "Unabled to update stored ingresses resource due to: %v", err)
+
+			// configure context to enable ingress class resource
+			ctxt.ingressClassResourceEnabled = true
+			ctxt.ingressClassResourceName = tests.IngressClassResourceName
+			ctxt.ingressClassControllerName = tests.IngressClassController
+			ctxt.ingressClassResourceDefault = false
```

Comment:
if default is true, would enabled=true still be necessary?

Reply:
`default=true` makes AGIC listen to any Ingress that doesn't have `ingressClassName` field set. AGIC does that only when `enabled=true`.